DMCompiler/Compiler/DM/DMLexer.cs

```diff
@@ -30,6 +30,11 @@ public sealed class DMLexer : TokenLexer {
             "..."
         };
 
+        private static readonly List<TokenType> ValidIdentifierComponents = new(2) {
+            TokenType.DM_Preproc_Identifier,
+            TokenType.DM_Preproc_Number
```

Comment:
Does checking 2 types in 1 spot really justify moving this to a list?

Reply:
Not really, but I was thinking this would be a much better place to handle the "operator[]" nonsense that I did kinda hackily in the operator overloads PR.

Which is to say, I intend to add a bunch more things to this list.